src/libcore/mem.rs

```diff
@@ -611,3 +611,38 @@ pub const POST_DROP_USIZE: usize = POST_DROP_U64 as usize;
 pub unsafe fn transmute_copy<T, U>(src: &T) -> U {
     ptr::read(src as *const T as *const U)
 }
+
+/// Returns the value of the discriminant for the enum variant in `v`.
+///
+/// The returned discriminant is cast to `u64`, no matter what the actual
```

Comment:
Added an example.

> Also what stability guarantees do the returned values have.

Hmm... this is a tough one. Since the discriminant depends on the order of enum variants, stability would imply that libraries following semver may not shuffle the order of variants, or add new ones except at the end, in minor versions (actually, they already can't add new variants, unless they have a hack like `std::io::ErrorKind::__Nonexhaustive`). This seems OK to me (needs to be documented of course), but others might have thoughts?